hypothesis-python/src/hypothesis/internal/lambda_sources.py

```diff
@@ -0,0 +1,424 @@
+            for candidate in local_lambdas:
+                if reflection.ast_arguments_matches_signature(
+                    candidate.args, sig
+                ) and _lambda_code_matches_node(f, candidate):
+                    return format_lambda(ast.unparse(candidate.body))
+
+        # Local parse failed or didn't produce a match, go ahead with the full parse
+        try:
+            tree = ast.parse("".join(source_lines))
+        except SyntaxError:  # pragma: no cover
+            all_lambdas = []
+        else:
+            all_lambdas = extract_all_lambdas(tree)
+        AST_LAMBDAS_CACHE[source_lines] = all_lambdas
+
+    aligned_lambdas = []
+    for candidate in all_lambdas:
+        if (
+            candidate.lineno - leeway <= lineno0 + 1 <= candidate.lineno + leeway
+            and reflection.ast_arguments_matches_signature(candidate.args, sig)
+        ):
+            aligned_lambdas.append(candidate)
+
+    aligned_lambdas.sort(key=lambda c: abs(lineno0 + 1 - c.lineno))
+    for candidate in aligned_lambdas:
+        if _lambda_code_matches_node(f, candidate):
+            return format_lambda(ast.unparse(candidate.body))
+
+    # None of the aligned lambdas match perfectly in generated code.
+    if (
+        fail_if_confused_with_perfect_candidate
+        and aligned_lambdas
+        and aligned_lambdas[0].lineno == lineno0 + 1
+    ):  # pragma: no cover
+        # This arg is forced on in conftest.py, to ensure we resolve all known
+        # cases.
+        raise ValueError("None of the source-file lambda candidates were matched")
+    return if_confused
+
+
+def lambda_description(f):
+    """
+    Returns a syntactically-valid expression describing `f`. This is often, but
+    not always, the exact lambda definition string which appears in the source code.
+    The difference comes from parsing the lambda ast into `tree` and then returning
+    the result of `ast.unparse(tree)`, which may differ in whitespace, double vs
+    single quotes, etc.
+
+    Returns a string indicating an unknown body if the parsing gets confused in any way.
+    """
+    try:
+        return LAMBDA_DESCRIPTION_CACHE[f]
+    except KeyError:
+        pass
+
+    key = _function_key(f, bounded_size=True)
+    failed_fnames = []
+    try:
+        description, failed_fnames = LAMBDA_DIGEST_DESCRIPTION_CACHE[key]
+        if (
+            "<unknown>" not in description and f.__code__.co_filename in failed_fnames
+        ):  # pragma: no cover
+            # Only accept the <unknown> description if it comes from parsing this
+            # file - otherwise, try again below, maybe we have more luck in another
+            # file. Once lucky, we keep keep using the successful description for *new*
+            # lambdas but not for this one - so it doesn't change name during its
+            # lifetime.
+            LAMBDA_DESCRIPTION_CACHE[f] = description
+            return description
+    except KeyError:
+        failed_fnames = []
```

Comment:
I think this can be replaced with `pass`, since the only source of a keyerror is from `LAMBDA_DIGEST_DESCRIPTION_CACHE[key]`?

Reply:
Yes that looks right. I'll rearrange as `try`-`except`-`else` to make that obvious.